src/core/reference/src/op/istft.cpp

```diff
@@ -0,0 +1,135 @@
+    std::vector<float> window_sum(batch_size * signal_length);
+
+    for (size_t batch = 0, batch_in_start = 0, batch_out_start = 0; batch < batch_size; ++batch) {
+        for (size_t frame_idx = 0; frame_idx < num_frames; ++frame_idx) {
+            const auto in_frame_start = batch_in_start + frame_idx * fft_out_shape_size;
+            const auto in_frame_end = in_frame_start + fft_out_shape_size;
+
+            const auto out_frame_start = batch_out_start + frame_idx * frame_step;
+            const auto out_frame_end = out_frame_start + frame_size;
+
+            std::vector<float> frame_data(data_t.data() + in_frame_start, data_t.data() + in_frame_end);
+            std::vector<float> frame_signal(frame_size);
+
+            reference::irdft(frame_data,
+                             fft_out_shape,
+                             {0},
+                             frame_signal.data(),
+                             frame_size_dim_shape_out,
+                             frame_size_dim_shape,
+                             frame_size);
+
+            std::transform(frame_signal.begin(),
+                           frame_signal.end(),
+                           mid_result.begin() + out_frame_start,
+                           mid_result.begin() + out_frame_start,
+                           std::plus<float>());
+
+            std::transform(window_sum.begin() + out_frame_start,
+                           window_sum.begin() + out_frame_end,
+                           pad_window.begin(),
+                           window_sum.begin() + out_frame_start,
+                           std::plus<float>());
+        }
+
+        if (normalized) {
+            std::transform(result + batch_out_start,
+                           result + batch_out_start + signal_length,
+                           result + batch_out_start,
+                           [sqrt_frame_size](float a) {
+                               return a * sqrt_frame_size;
+                           });
+        }
+
+        std::transform(result + batch_out_start,
+                       result + batch_out_start + signal_length,
+                       window_sum.begin(),
+                       result + batch_out_start,
+                       [](float a, float b) {
+                           if (b != 0.f)
+                               return a / b;
+                           else
+                               return 0.f;
+                       });
+
+        if (center) {
+            const int64_t margin = (frame_size / 2);
+            const size_t result_start = batch_out_start + margin;
+            const int64_t data_end = signal_length - (frame_size / 2);
+            int64_t signal_end = final_signal_length < data_end ? final_signal_length : data_end;
+            std::copy(result + result_start,
+                      result + result_start + signal_end,
+                      final_result + (batch * final_signal_length));
```

Comment:
```suggestion
            final_result  = std::copy(result + result_start,
                      result + result_start + signal_end,
                      final_result);
```
It can be reused in other cases that return value from copu, transform will update the destination iterator and there will be no need to calculate offsets explicity.

Reply:
In this case, not always. The next `final_result` start pointer doesn't need to be equal to the end of the previous result.
This is because the provided "signal_length" can be bigger than the actual size of the calculated data and the `final_result` is filled with zeros at the end.
But I've updated some pointers to increment them at the end of the loop instead.